workitem/link/link_repository_blackbox_test.go

```diff
@@ -61,6 +63,220 @@ func (s *linkRepoBlackBoxTest) TestList() {
+		require.NoError(t, err)
+
+		// List all workitem children after reordering. Again, List returns workitems in descending order
+		// afterReorder returns
+		// Child1 (2000)
+		// Child2 (1500)
+		afterReorder, _, err := s.workitemLinkRepo.ListWorkItemChildren(s.Ctx, fxt.WorkItems[0].ID, nil, nil)
+		require.NoError(t, err)
+		require.Len(t, afterReorder, 2)
+
+		var expectedOrder []interface{}
+		for i := 1; i < 3; i++ {
+			expectedOrder = append(expectedOrder, fxt.WorkItems[i].ID) // expectedOrder = [child1, child2]
+		}
+		for i, v := range afterReorder {
+			assert.Equal(t, v.ID, expectedOrder[i])
+		}
+	})
+}
+
+// Tests reorder of child work item **top**
+func (s *linkRepoBlackBoxTest) TestChildrenReorderTop() {
+	s.T().Run("ok - child work items reorder top", func(t *testing.T) {
+		// Create 3 workitems -> Parent, Child 1 & 2
+		// Workitems in ascending order =
+		// Parent (1000)
+		// Child1 (2000)
+		// Child2 (3000)
+		fxt := tf.NewTestFixture(t, s.DB,
+			tf.Identities(1),
+			tf.Spaces(1),
+			tf.WorkItems(3),
+			tf.WorkItemLinkTypes(1, func(fxt *tf.TestFixture, idx int) error {
+				fxt.WorkItemLinkTypes[idx].ForwardName = "parent of"
+				return nil
+			}),
+			tf.WorkItemLinks(2, func(fxt *tf.TestFixture, idx int) error {
+				fxt.WorkItemLinks[idx].SourceID = fxt.WorkItems[0].ID
+				fxt.WorkItemLinks[idx].TargetID = fxt.WorkItems[idx+1].ID
+				return nil
+			}),
+		)
+
+		// List all children before reordering.
+		// List returns workitems in descending order.
+		// So, beforeReorder returns
+		// Child2 (3000)
+		// Child1 (2000)
+		beforeReorder, _, err := s.workitemLinkRepo.ListWorkItemChildren(s.Ctx, fxt.WorkItems[0].ID, nil, nil)
+		require.NoError(t, err)
+		require.Len(t, beforeReorder, 2)
+
+		// Reorder child 1 to topmost position
+		s.workitemRepo.Reorder(s.Ctx, fxt.Spaces[0].ID, workitem.DirectionTop, nil, *fxt.WorkItems[1], fxt.Identities[0].ID)
+
+		// List all workitem children after reordering. Again, List returns workitems in descending order
+		// afterReorder returns
+		// Child1 (4000)
+		// Child2 (3000)
+		afterReorder, _, err := s.workitemLinkRepo.ListWorkItemChildren(s.Ctx, fxt.WorkItems[0].ID, nil, nil)
+		require.NoError(t, err)
+		require.Len(t, afterReorder, 2)
+
+		var expectedOrder []interface{}
+		for i := 1; i < 3; i++ {
+			expectedOrder = append(expectedOrder, fxt.WorkItems[i].ID) // expectedOrder = [child1, child2]
+		}
+		for i, v := range afterReorder {
+			assert.Equal(t, v.ID, expectedOrder[i])
+		}
+	})
+}
+
+// Tests reorder of child work item **bottom**
+func (s *linkRepoBlackBoxTest) TestChildrenReorderBottom() {
+	s.T().Run("ok - child work items reorder bottom", func(t *testing.T) {
+		// Create 3 workitems -> Parent, Child 1 & 2
+		// Workitems in ascending order =
+		// Parent (1000)
+		// Child1 (2000)
+		// Child2 (3000)
+		fxt := tf.NewTestFixture(t, s.DB,
```

Comment:
@DhritiShikhar why do you not use subtests and why do you keep repeating the test fixture definition over and over again?